graphql/query_resolver.go

```diff
@@ -184,17 +184,30 @@ func (r *queryResolver) Distros(ctx context.Context, onlySpawnable bool) ([]*res
 func (r *queryResolver) DistroTaskQueue(ctx context.Context, distroID string) ([]*restModel.APITaskQueueItem, error) {
 	distroQueue, err := model.LoadTaskQueue(distroID)
 	if err != nil {
-		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error getting task queue for distro %v: %v", distroID, err.Error()))
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("getting task queue for distro '%v': %v", distroID, err.Error()))
 	}
 	if distroQueue == nil {
 		return nil, ResourceNotFound.Send(ctx, fmt.Sprintf("cannot find queue with distro ID `%s`", distroID))
 	}
 
+	idToIdentifierMap := map[string]string{}
 	taskQueue := []*restModel.APITaskQueueItem{}
 
 	for _, taskQueueItem := range distroQueue.Queue {
 		apiTaskQueueItem := restModel.APITaskQueueItem{}
+
+		projectID := taskQueueItem.Project
+
+		if _, ok := idToIdentifierMap[projectID]; !ok {
+			identifier, err := model.GetIdentifierForProject(projectID)
```

Comment:
I _think_ there are cases where there _is_ no identifier, for example with untracked branches (since these point to a "hidden" project with no identifier). It might be better then to store the identifier outside of the map in a variable (maybe instead of the projectID variable since taskQueueItem.Project works just as well), and then only add to map and assign to this variable if identifier is non empty

Reply:
Oh yeah I definitely forgot about that. I think in that case the identifier will be an empty string — and then on the UI we can just do `projectIdentifier || projectID` (i.e. show project identifier if non-empty, otherwise show project ID). Like this:

<img width="716" alt="Screenshot 2024-06-05 at 10 48 48 PM" src="https://github.com/evergreen-ci/evergreen/assets/47064971/2e48caaf-0e73-4f0d-9362-e265720968fa">

Does that work? Technically I think it's correct that they would have an empty project identifier. 

I can get rid of the `projectID` variable though, I just added it to help reading